pydantic_numpy/model.py

```diff
@@ -152,12 +163,20 @@ def dump(
 
         return dump_directory_path
 
-    def _dump_numpy_split_dict(self) -> tuple[dict, dict]:
-        ndarray_field_to_array = {}
-        other_field_to_value = {}
-
-        for k, v in self.model_dump(exclude_unset=True).items():
-            if isinstance(v, np.ndarray):
+    @staticmethod
+    def _dump_numpy_split_dict(model: "NumpyModel") -> tuple[dict[str, np.ndarray], dict[str, Any]]:
+        ndarray_field_to_array = dict[str, np.ndarray]()
+        other_field_to_value = dict[str, Any]()
+
+        for k, v in model.model_dump(exclude_unset=True).items():
+            item = getattr(model, k)
+            if isinstance(item, NumpyModel):
+                submodel_ndarray_field_to_array, submodel_other_field_to_value = item._dump_numpy_split_dict(item)
+                for sub_k_np, sub_v_np in submodel_ndarray_field_to_array.items():
+                    ndarray_field_to_array[f"{k}/{sub_k_np}"] = sub_v_np
+                for sub_k_other, sub_v_other in submodel_other_field_to_value.items():
+                    other_field_to_value[f"{k}/{sub_k_other}"] = sub_v_other
```

Comment:
Some spacing between different components:
```suggestion
                    other_field_to_value[f"{k}/{sub_k_other}"] = sub_v_other

```